### PR TITLE
Modify NVRTC compiler arguments for profiling workflow

### PR DESCRIPTION
**Pull request by karen-sy**
Warp uses NVRTC to compile a kernel, generated via `codegen.py` from python source code, during runtime. Since the source code for nvrtc compilation is a raw string and not a file, NSight Compute cannot associate the kernel with any source file. Note that, if we were using NVCC, `--generate-line-info` suffices to associate the program to its source file's code line-by-line, but in our case, given no filename to associate, the profiler can't do anything with the line number information.

This PR enables local build of Warp on CUDA 12.6, and modifies `nvrtcCreateProgram` such that the kernel is (1) always compiled with `--generate-line-info`, (2) and associated with a `module_codegen.cu`, which is a cache file that Warp copies the raw code string into (and invokes, via a utf-8 encoded function signature, during execution context). 

In short, a Warp kernel is now compiled in a way such that profilers like Nsight Compute can generate source-to-assembly comparison views:
image

**Remaining TODOs:**
 - Write function that returns cache directory name for a given Warp kernel. (Should be some deterministic function of the module/file/kernel name, and one or more of those names' utf-8 encoding.) With this functionality, we can automate the `ncu` workflow to recognize the `--source-folders` without having to compile the kernel first to initialize the cache directory. See the print statement in `context.py`
 - Make "Profiling Mode" a field in `config.py` such that `--generate-line-info` and potential future customizations can be toggled easily in one argument.

### CODE REVIEW

**Comment by karen-sy** [comment]
All remaining TODO's are resolved natively with existing Warp functionalities, see `profile.sh` for an example run.
